### ISSUE TITLE
Undefined Index: Instructions when saving supertable in matrix

### ISSUE THREAD

**Issue by RobErskine** [engram-design/SuperTable#89]
Hey there,

I'm getting an error when I try and save a SuperTable (I am running v1.0.1) field inside of a pretty complex matrix. I get the following error:
```
Undefined Index: Instructions
/var/www/vhost/jh-different-world/staging/plugins/SuperTable/supertable/fieldtypes/SuperTableFieldType.php(79)
```

[Full stack trace here in this screenshot.](https://cl.ly/2I0O3V441t2p/Screen%20Shot%202016-09-20%20at%2011.32.45%20AM.jpg)

For what it's worth, I have another SuperTable field inside of another matrix block. So there are now two SuperTable fields, and the error is happening on the second one. Am I missing something silly?

**Comment by engram-design** [comment]
Seems a little odd, but just to confirm, you've got a Matrix > Super Table field layout (not Super Table > Matrix > Super Table?) Is it possible to send through a screenshot before you're about to save? Just in case there are any clues there...

**Comment by RobErskine** [comment]
Yup, you are correct. I have a Matrix > Block Type > SuperTable field inside the block.

I've attached a gif of me tabbing through the two different blocks and then trying to save them out.

![Super Table Error](https://cl.ly/0a3q2J19172E/Screen%20Recording%202016-09-21%20at%2012.56%20PM.gif)

When I refresh the page or go back to the field settings, my fields aren't saved, and only one supertable instance exists.

**Comment by engram-design** [comment]
Thanks for the gif - thats super helpful. Let me investigate and see where we get to!

**Comment by RobErskine** [comment]
For sure! FWIW there is actually a _third_ supertable instance within this matrix. I'm not sure if that would cause any more problems than there only being two.

![3rd super table box](https://cl.ly/383W000q4030/Screen%20Shot%202016-09-22%20at%2010.13.39%20AM.jpg)

**Comment by engram-design** [comment]
The only thing that might be of interest is if there are any Super Table fields with the same handle?

**Comment by RobErskine** [comment]
Hm I don't know if that's it. They're all unique.

- externalLinks
  - linkLabel (plaintext)
  - linkDestination (plaintext)
- galleryImages
  - galleryItemTitle (plaintext)
  - galleryItemDescription (simple richtext)
  - galleryItemImage (asset - image)
- timelineCards
  - timelineYear (number)
  - timelineColor (color hex)
  - timelineDescription (simple richtext)
  - timeline X (number)
  - timeline Y (number)

Could it be that the instruction field on some of those is empty? Or that I'm using too many plaintexts or something?

**Comment by engram-design** [comment]
I've tried to reproduce this to no avail I'm afraid. But you raise an interesting point - I wonder if its something to do with `max_input_vars` or `post_max_size`, being such a large collection of fields. What makes me say this, is your initial post mentioned not being able to find `Instructions` index, while your screengrab complains about `type` index.

Anyway - have reached out to you on Slack to hopefully get to the bottom of this.

**Comment by RobErskine** [comment]
Sorry for the tardy response, been out of town the last few days.

BUT, I did finally get around to addressing this, and you were correct, it had to do with `max_input_vars` being too low.

For anyone else suffering from a massive matrix / super table config, edit your `php.ini` to change `max_input_vars = 1000` to something else higher. For me, I changed mine to `max_input_vars = 10000`. Whether this is bad practice or not, I guess I'll find out in a couple of days if anything else goes down😛 .
